React/Base/RCTBridge.m

```diff
@@ -84,8 +84,7 @@ typedef NS_ENUM(NSUInteger, RCTBridgeFields) {
     RCTModuleNamesByID = [NSMutableArray array];
 
     RCTEnumerateClasses(^(__unsafe_unretained Class cls) {
-      if ([cls conformsToProtocol:@protocol(RCTBridgeModule)]) {
-
+      if (class_getClassMethod(cls, @selector(bridgeModule)) != NULL) {
```

Comment:
Yup, that's correct. It is theoretically not a problem, because at run time frameworks check for the existence of methods, not of conformance to a protocol as a whole.